trin-history/src/jsonrpc.rs

```diff
@@ -1,245 +1,304 @@
+            let network = self.network.clone();
+            tokio::spawn(async move { complete_request(network, request).await });
+        }
+    }
+}
 
-                    let _ = request.resp.send(response);
-                }
-                HistoryEndpoint::RecursiveFindNodes(node_id) => {
-                    let node_id = discv5::enr::NodeId::from(node_id.0);
-                    let mut nodes = self.network.overlay.lookup_node(node_id).await;
-                    nodes.sort_by(|a, b| {
-                        XorMetric::distance(&node_id.raw(), &a.node_id().raw())
-                            .cmp(&XorMetric::distance(&node_id.raw(), &b.node_id().raw()))
-                    });
-                    let nodes: Vec<Enr> = nodes.into_iter().take(16).collect();
-                    let response = Ok(json!(nodes));
-                    let _ = request.resp.send(response);
-                }
-            }
+// Generates a response for a given request and sends it to the receiver.
+async fn complete_request(network: Arc<RwLock<HistoryNetwork>>, request: HistoryJsonRpcRequest) {
+    let response: Result<Value, String> = match request.endpoint {
+        HistoryEndpoint::LocalContent(content_key) => local_content(network, content_key).await,
+        HistoryEndpoint::PaginateLocalContentKeys(offset, limit) => {
+            paginate_local_content_keys(network, offset, limit).await
+        }
+        HistoryEndpoint::Store(content_key, content_value) => {
+            store(network, content_key, content_value).await
+        }
+        HistoryEndpoint::RecursiveFindContent(content_key) => {
+            recursive_find_content(network, content_key, false).await
+        }
+        HistoryEndpoint::TraceRecursiveFindContent(content_key) => {
+            recursive_find_content(network, content_key, true).await
+        }
+        HistoryEndpoint::DataRadius => {
+            let radius = network.read().await.overlay.data_radius();
+            Ok(json!(*radius))
+        }
+        HistoryEndpoint::FindContent(enr, content_key) => {
+            find_content(network, enr, content_key).await
         }
+        HistoryEndpoint::FindNodes(enr, distances) => find_nodes(network, enr, distances).await,
+        HistoryEndpoint::Gossip(content_key, content_value) => {
+            gossip(network, content_key, content_value).await
+        }
+        HistoryEndpoint::Offer(enr, content_key, content_value) => {
+            offer(network, enr, content_key, content_value).await
+        }
+        HistoryEndpoint::Ping(enr) => ping(network, enr).await,
+        HistoryEndpoint::RoutingTableInfo => Ok(bucket_entries_to_json(
+            network.read().await.overlay.bucket_entries(),
+        )),
+        HistoryEndpoint::RecursiveFindNodes(node_id) => {
+            recursive_find_nodes(network, node_id).await
+        }
+    };
+    let _ = request.resp.send(response);
+}
+
+// Constructs a JSON call for the method.
```

Comment:
Could we change those comments to doc comments and additional info for the endpoint, something like:

```suggestion
/// Constructs a JSON call for the RecursiveFindContent method.
```

Reply:
done